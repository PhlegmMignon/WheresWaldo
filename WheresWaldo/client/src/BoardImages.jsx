import convention from "./assets/nMpQXwq.jpg";
import convention_kon from "./assets/kon.webp";
import convention_luffy from "./assets/luffy.png";
import convention_zim from "./assets/Zim.yelling.svg";

import stage2 from "./assets/stage3.png";
import stage2_fish from "./assets/fish.png";
import stage2_cannon from "./assets/cannon.png";
import stage2_egg from "./assets/egg.png";

export default function BoardImages() {
  const getImages = () => {
    return [
      {
        name: "convention",
        src: convention,
        id: 0,
        characters: [
          { name: "zim", src: convention_zim, id: 0 },
          { name: "luffy", src: convention_luffy, id: 1 },
          { name: "kon", src: convention_kon, id: 2 },
        ],
      },
      {
        name: "stage 2",
        src: stage2,
        id: 1,
        characters: [
          { name: "fish", src: stage2_fish, id: 0 },
          { name: "cannon", src: stage2_cannon, id: 1 },
          { name: "egg", src: stage2_egg, id: 2 },
        ],
      },
    ];
  };

  return { getImages };
}
