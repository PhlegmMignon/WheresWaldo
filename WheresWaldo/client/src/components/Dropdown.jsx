export default function Dropdown({ gameImage, setFoundStatus, coordinate }) {
  // useEffect(() => {
  //   const ele = document.getElementById("charList");
  //   for (const child of ele.children) {
  //     const id = child.id.charAt(child.id.length - 1);
  //     child.addEventListener("click", setFoundStatus(e, id, coordinate));
  //   }
  // }, [gameImage]);

  const handleClick = (e) => {
    e.stopPropagation();
    const id = e.target.id.charAt(e.target.id.length - 1);
    // console.log(coordinate);
    setFoundStatus(
      gameImage.name,
      gameImage.characters[id].name,
      coordinate,
      id
    );
  };

  return (
    <div
      id="dropdown"
      className="absolute rounded-lg"
      style={{ left: coordinate[0] + "px", top: coordinate[1] - 110 + "px" }}
    >
      <li className="list-none flex flex-col p-2 gap-2 " id="charList">
        {gameImage.characters.map((char) => {
          return (
            <ul key={char.id} className="flex justify-center">
              <img
                id={"char" + char.id}
                onClick={handleClick}
                className="h-16 w-auto"
                src={char.src}
                alt=""
              />
            </ul>
          );
        })}
      </li>
    </div>
  );
}
