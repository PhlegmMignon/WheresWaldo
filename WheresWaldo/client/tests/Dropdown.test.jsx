import { render, screen } from "@testing-library/react";
import { vi } from "vitest";
import userEvent from "@testing-library/user-event";
import Dropdown from "../src/components/Dropdown.jsx";
import BoardImages from "../src/BoardImages.jsx";

describe("Dropdown unit tests", () => {
  it("Renders dropdown", async () => {
    let gameImage = BoardImages().getImages();

    const { container } = render(
      <Dropdown coordinate={[1, 1]} gameImage={gameImage[0]} />
    );

    expect(container).toMatchSnapshot();
  });
});

describe("Dropdown integration tests", () => {
  it("Clicking char image calls handleClick", async () => {
    const user = userEvent.setup();
    let gameImage = BoardImages().getImages();
    const fakeSetFoundStatus = vi.fn();
    const fakeSetModal = vi.fn();

    render(
      <Dropdown
        coordinate={[1, 1]}
        gameImage={gameImage[0]}
        setFoundStatus={fakeSetFoundStatus}
        setModal={fakeSetModal}
      />
    );
    let btn = screen.getByRole("img", { name: "image of luffy" });

    await user.click(btn);
    expect(fakeSetFoundStatus).toHaveBeenCalled();
    expect(fakeSetModal).toHaveBeenCalled();
  });
});
