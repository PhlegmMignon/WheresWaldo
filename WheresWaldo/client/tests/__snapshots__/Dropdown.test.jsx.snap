// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Dropdown unit tests > Renders dropdown 1`] = `
<div>
  <div
    class="absolute rounded-lg"
    id="dropdown"
    style="left: 1px; top: -109px;"
  >
    <li
      class="list-none flex flex-col p-2 gap-2 "
      id="charList"
    >
      <ul
        class="flex justify-center"
      >
        <img
          alt="image of zim"
          class="h-16 w-auto"
          id="char0"
          src="/src/assets/Zim.yelling.svg"
        />
      </ul>
      <ul
        class="flex justify-center"
      >
        <img
          alt="image of luffy"
          class="h-16 w-auto"
          id="char1"
          src="/src/assets/luffy.png"
        />
      </ul>
      <ul
        class="flex justify-center"
      >
        <img
          alt="image of kon"
          class="h-16 w-auto"
          id="char2"
          src="/src/assets/kon.webp"
        />
      </ul>
    </li>
  </div>
</div>
`;
